virttest/utils_net.py

```diff
@@ -4351,3 +4351,50 @@ def delete_ovs_bridge(ovs_bridge_name, session=None, ignore_status=False):
     return utils_misc.cmd_status_output(tmux_cmd, shell=True, verbose=True,
                                         ignore_status=ignore_status,
                                         session=session)
+
+
+def create_linux_bridge(linux_bridge_name, iface_name, ignore_status=False):
+    """
+    Create linux bridge and connect a physical interface to the bridge via tmux command on local host
+    1. use tmux to avoid the network broken;
+    2. add a physical interface to the bridge, and set the bridge up;
+
+    :param linux_bridge_name: The linux bridge
+    :param iface_name: the physical interface to be attached to the bridge
+    :param ignore_status: Whether to raise an exception when command fails
+    :return: bridge created or raise exception
+    """
+    # Create bridge
+    br_path = "/sys/class/net/%s" % linux_bridge_name
+    if not utils_package.package_install(['tmux', 'dhcp-client']):
+        raise exceptions.TestError("Failed to install the required packages.")
+    if os.path.exists(br_path):
+        raise exceptions.TestError("Create bridge error as there is already an interface named '%s' on the host"
```

Comment:
Is it possible to delete the existing bridge by calling delete_linux_bridge?

Reply:
yes, I have updated the function to be like delete_linux_bridge(linux_bridge_name, iface_name=None, ignore_status=False)
If there is physical interface attached to the bridge, the bridge name and interface name both need to be specified. The interface name is needed when we try to recover the connection.
If there is no physical interface attached to the bridge, only the bridge name is needed.

Reply:
Ok. You can call delete_linux_bridge🙂